test/helpers/optimisticPlugin.js

```diff
@@ -0,0 +1,37 @@
+const EventEmitter = require('events')
+
+class OptimisticPlugin extends EventEmitter {
```

Comment:
Would call that `MockOptimisticPlugin` and put it under `test/mocks/optimisticPlugin.js`.